includes/api/class-wc-rest-shipping-methods-controller.php

```diff
@@ -0,0 +1,221 @@
+				'callback'            => array( $this, 'get_items' ),
+                'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+        register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<id>[\w-]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				'args'                => array(
+					'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+				),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+    /**
+	 * Check whether a given request has permission to view system status.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+        if ( ! wc_rest_check_manager_permissions( 'shipping_methods', 'read' ) ) {
+        	return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+		return true;
+	}
+
+    /**
+	 * Check if a given request has access to read a shipping method.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'shipping_methods', 'read' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot view this resource.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+		return true;
+	}
+
+    /**
+	 * Get shipping methods.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+        $wc_shipping = WC_Shipping::instance();
+        $response    = array();
+        foreach ( $wc_shipping->get_shipping_methods() as $id => $shipping_method ) {
+            $method = $this->prepare_item_for_response( $shipping_method, $request );
+			$method = $this->prepare_response_for_collection( $method );
+			$response[] = $method;
+        }
+        return rest_ensure_response( $response );
+	}
+
+    /**
+     * Get a single Shipping Method.
+     *
+     * @param WP_REST_Request $request
+     * @return WP_REST_Response|WP_Error
+     */
+    public function get_item( $request ) {
+        $wc_shipping = WC_Shipping::instance();
+        $methods     = $wc_shipping->get_shipping_methods();
+        if ( empty ( $methods[ $request['id'] ] ) ) {
+            return new WP_Error( 'woocommerce_rest_shipping_method_invalid', __( "Resource doesn't exist.", 'woocommerce' ), array( 'status' => 404 ) );
```

Comment:
This one should be OK - Double quotes should be used when there is a single quote in the string (instead of escaping). This also matches the error some other endpoints return.